.github/container/gcp-autoconfig.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+if [ "${DISABLE_GCP_TCPX_SETUP}" == 1 ]; then 
+  return
+fi
+
+# Colors
+GREEN='\033[0;32m'
+NOCOLOR='\033[0m'
+
+# If google.internal exists, we are most likely running on GCP.
+if host metadata.google.internal &> /dev/null; then
+    echo -e "${GREEN}
+============================== JAX-ToolBox on GCP ==============================
+
+It looks like you're running on GCP. In order to maximize your multi-node
+performance, you'll need to use Google's TCPx NCCL plugin. This container ships
+the plugin at ${TCPX_LIBRARY_PATH}, which is already added to LD_LIBRARY_PATH by
+this script $2.
+
+For A3 instances, this script also automatically determines the best NICs to use
+by NCCL for maximum performance.
+
+However, there are additional steps you will need to take. Mainly, you'll need
+to run a separate receive-datapath-manager daemon on each of your nodes, and
+correctly configure your networks and NICs. For more information, please see the
+guide at: https://cloud.google.com/compute/docs/gpus/gpudirect#provide-access
```

Comment:
We should also info users that if they are running on DGX, this is already done for them.

Reply:
Issue created for follow-up: https://github.com/NVIDIA/JAX-Toolbox/issues/722